src/Domain/FunctionsOfTime/PiecewisePolynomial.cpp

```diff
@@ -63,13 +70,45 @@ PiecewisePolynomial<MaxDeriv>::func_and_derivs(const double t) const noexcept {
 
 template <size_t MaxDeriv>
 void PiecewisePolynomial<MaxDeriv>::update(
-    const double time_of_update, DataVector updated_max_deriv) noexcept {
+    // Clang-tidy says to use 'const DataVector& updated_max_deriv'.
+    // However, updated_max_deriv is std::moved out of inside this function.
+    // NOLINTNEXTLINE(performance-unnecessary-value-param)
+    const double time_of_update, DataVector updated_max_deriv,
+    const double expiration_time) noexcept {
   if (time_of_update <= deriv_info_at_update_times_.back().time) {
     ERROR("t must be increasing from call to call. "
           << "Attempted to update at time " << time_of_update
           << ", which precedes the previous update time of "
           << deriv_info_at_update_times_.back().time << ".");
   }
+  if (expiration_time <= expiration_time_) {
+    ERROR("expiration_time must be nondecreasing from call to call. "
```

Comment:
Either "<=" -> "<" or "nondecreasing" -> "increasing".

The current check does not match the one in `reset_expiration_time`.  I don't know if that is intentional.